Response List: Fixed bug - The code for validation responses can now handle `list`

dvaerum opened the pull request:
I run into a bug when writing unittests there validated the returned JSON-object which contains a `list`.

I created 2 unittests to verify that my bug fix works :D

Checklist:

- [x] Run `pytest tests` and no failed.
- [x] Run `flake8 flask_openapi3 tests examples` and no failed.
- [x] Run `mkdocs serve` and no failed.

luolingchun commented:
JSON-object contain not only `list` but also `dict`, like:

```python
class SelloutList(BaseModel):
    __root__: List[Sellout]


class SelloutDict(BaseModel):
    __root__: Dict[str, Sellout]


class SelloutDict2(BaseModel):
    __root__: Dict[Any, Any]
```

So there is a better solution, using [custom_root_type](https://github.com/luolingchun/flask-openapi3/blob/master/flask_openapi3/do_wrapper.py#L83):

```python
if body.__custom_root_type__:
    # https://pydantic-docs.helpmanual.io/usage/models/#custom-root-types
    body_ = body(__root__=obj)
else:
    body_ = body(**obj)
```

dvaerum commented:
Hey @luolingchun, I see what you mean and yes, it is a better solution. I have updated the code and the unittests I added to the library are still happy, so I will say everything looks good :D
